classes/class.pmproemail.php

```diff
@@ -1306,26 +1306,70 @@ function sendTrialEndingEmail( $user = NULL, $membership_id = NULL )
 			return $this->sendEmail();
 		}
 		
-		
-		function sendMembershipExpiredEmail( $user = NULL, $membership_id = NULL )
-		{
+		/**
+		 * Send the member an email when their membership has ended.
+		 *
+		 * @param object $user The WordPress user object.
+		 * @param int $membership_id The member's membership level ID.
+		 * @return bool Whether the email was sent successfully.
+		 * @since TBD
+		 */
+		function sendMembershipExpiredEmail( $user = NULL, $membership_id = NULL ) {
 			global $current_user;
-			if(!$user)
+			if( !$user ) {
 				$user = $current_user;
-			
-			if(!$user)
-				return false;						
+			}
+			//Bail if still we don't have a user.
+			if( !$user ) {
+				return false;
+			}
+			//If we have a membership ID, get the level.
+			if (! empty( $membership_id ) ) {
+				$membership_level = pmpro_getLevel( $membership_id );
+			//If not let's find out what's the last ended membership level.
+			} else {
+				//Get all levels for a specific member
+				$all_levels = pmpro_getMembershipLevelsForUser( $user->ID );
+				//Bail if we don't have any levels.
+				if ( empty( $all_levels ) ) {
+					return false;
+				}
+				//get the older membership
+				$membership_level = array_reduce( $all_levels, function ( $oldest_level, $current_level ) {
+					// Check if the level status is not "active" and compare end dates
+					if ( $current_level->enddate !== null && ( $oldest_level === null || $current_level->enddate > $oldest_level->enddate ) ) {
+						return $current_level;
+					}
+					return $oldest_level;
+				});
```

Comment:
The level status is not currently being checked here. `$current_level->enddate` may be filled even if the membership is still active.

`$current_level->enddate` is either the date that a membership was ended if the membership is not active, or is the expiration date for the membership if the membership is still active.

Reply:
why would be the best way to check status then ? compare enddate  with now() function  besides the null scenario ?

Reply:
Pushed a commit with a direct SQL query instead of using pmpro_getMembershipLevelsForUser(). Should be more efficient and cleaner to read